src/SIM/SIMbase.C

```diff
@@ -704,7 +705,12 @@ bool SIMbase::updateGrid (const Vector& displ)
   Vector locdisp;
   for (size_t i = 0; i < myModel.size() && ok; i++)
   {
-    myModel[i]->extractNodeVec(displ,locdisp,myModel[i]->getNoSpaceDim(),-1);
+    if (this->mixedProblem())
+      this->extractPatchSolution(displ, locdisp, myModel[i],
+                                 nsd, ASMmxBase::geoBasis);
```

Comment:
In case of the very unlikely event that patches have different number of space dimensions, `nsd` should perhaps be replaced by `myModel[i]->getNoSpaceDim()`, like in the extractNodeVec call? Mostly for consistency.